src/Nethermind/Nethermind.Core2/SquareRootExtensions.cs

```diff
@@ -0,0 +1,37 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+namespace Nethermind.Core2
+{
+    public static class UInt64Extensions
+    {
+        /// <summary>
+        /// Return the largest integer x such that x**2 is less or equal n.
+        /// </summary>
+        public static ulong SquareRoot(this ulong value)
```

Comment:
This is not actually the square root, so I would avoid calling it that. i.e. square root of 8 is 2.828427...

The method in specs is called IntegerSquareRoot(), so maybe continue to use that. IntegerSquareRoot(8) = 2.

Reply:
ok